kong/router.lua

```diff
@@ -260,19 +260,12 @@ local function marshall_route(r)
 
   -- TODO: service.path is not supported in new model
   local path = service.path or null
-  local file = path
-  if path ~= null then
-    if sub(path, -1) == "/" and path ~= "/" then
-      file = sub(path,  1, -2)
-    end
 
-  else
+  if path == null then
```

Comment:
my assumption here was that the TODO comment above indicated that in the future this value comes in as `null` and not `nil`, hence I retained the existing logic with the extra check. And it seemed a bad combination to also fix that here?

Reply:
There was once a time when new dao returned `null` instead of `nil`, but that was changed to configurable. Today it returns `nil` by default. Only the admin API uses `null`.

Reply:
Fixed after consulting @bungle, see https://github.com/Kong/kong/pull/3780/commits/d1cad14432c14801d23280ea3f6bace5f9186ea3

Reply:
Yes, just like @bungle said. Thank you!